public/app/features/dimensions/editors/ScalarDimensionEditor.tsx

```diff
@@ -0,0 +1,110 @@
+import { InlineField, InlineFieldRow, RadioButtonGroup, Select, useStyles2 } from '@grafana/ui';
+import { useFieldDisplayNames, useSelectOptions } from '@grafana/ui/src/components/MatchersUI/utils';
+import { NumberInput } from './NumberInput';
+import { css } from '@emotion/css';
+
+const fixedValueOption: SelectableValue<string> = {
+  label: 'Fixed value',
+  value: '_____fixed_____',
+};
+
+const scalarOptions = [
+  { label: 'Mod', value: ScalarDimensionMode.Mod, description: 'Use values, mod from max' },
+  { label: 'Capped', value: ScalarDimensionMode.Capped, description: 'Use field values, capped to max and min' },
+];
+
+export const ScalarDimensionEditor: FC<StandardEditorProps<ScalarDimensionConfig, ScalarDimensionOptions, any>> = (
+  props
+) => {
+  const { value, context, onChange } = props;
+
+  const DEFAULT_VALUE = 0;
+
+  const fieldName = value?.field;
+  const isFixed = Boolean(!fieldName);
+  const names = useFieldDisplayNames(context.data);
+  const selectOptions = useSelectOptions(names, fieldName, fixedValueOption, FieldType.number);
+
+  const styles = useStyles2(getStyles);
+
+  const onSelectChange = useCallback(
+    (selection: SelectableValue<string>) => {
+      const field = selection.value;
+      if (field && field !== fixedValueOption.value) {
+        onChange({
+          ...value,
+          field,
+        });
+      } else {
+        const fixed = value.fixed ?? DEFAULT_VALUE;
+        onChange({
+          ...value,
+          field: undefined,
+          fixed,
+        });
+      }
+    },
+    [onChange, value]
+  );
+
+  const onModeChange = useCallback(
+    (mode) => {
+      onChange({
+        ...value,
+        mode,
+      });
+    },
+    [onChange, value]
+  );
+
+  const onValueChange = useCallback(
+    (v: number | undefined) => {
+      onChange({
+        ...value,
+        field: undefined,
+        fixed: v ?? DEFAULT_VALUE,
+      });
+    },
+    [onChange, value]
+  );
+
+  const val = value ?? {};
+  const mode = value?.mode ?? ScalarDimensionMode.Mod;
+  const selectedOption = isFixed ? fixedValueOption : selectOptions.find((v) => v.value === fieldName);
+  return (
+    <>
+      <div>
+        <InlineFieldRow>
+          <InlineField label="Limit" labelWidth={8} grow={true}>
+            <RadioButtonGroup value={mode} options={scalarOptions} onChange={onModeChange} fullWidth />
+          </InlineField>
+        </InlineFieldRow>
+        <Select
+          menuShouldPortal
+          value={selectedOption}
+          options={selectOptions}
+          onChange={onSelectChange}
+          noOptionsMessage="No fields found"
+        />
+      </div>
+      <div className={styles.range}>
+        {isFixed && (
+          <InlineFieldRow>
+            <InlineField label="Value" labelWidth={8} grow={true}>
+              <NumberInput value={val?.fixed ?? DEFAULT_VALUE} onChange={onValueChange} />
```

Comment:
We may want to consider limiting the `fixed` value to be within the given min / max values, else user can enter any number for fixed value (0-360 makes sense, but having option of entering in 10000 etc. may potentially be confusing to user)

Reply:
👍 sounds good. what are your thoughts on us adding that to another PR so we can share validation with other components using the NumberInput?

Reply:
Do you mean cleaning up number inputs across the board for geomap? Sure that sounds good